Switch to base image ubi8-minimal, use staged build pattern.

MaxCWhitehead opened the pull request:
- Uses **ubi8-minimal** instead of ubi8-modphp74.
- Use staged build pattern to discard build artifacts, we copy imagemagick build output to a fresh image. This Significantly reduces image size. Expecting image around **~276MB** from **~968MB.**
- **No longer have access to dnf** (only microdnf), and yum is not install, this will require modification to images using this base. 
- I **removed installation of composer**. I do not believe this belongs in final image, derived images requiring composer should install it, get dependencies, and then remove from their own image.
- **rpm-file-permissions** script is on longer present, I am not 100% sure what permissions we were resetting here, and if there are any concerns on file permissions in final image.

**Side note on ImageMagick dependency:**
We are installing ImageMagick application, and imagemagick as a php extension. I believe this is to support drupal module '[ImageMagick](https://www.drupal.org/project/imagemagick)' in dependent images. I believe this module only requires convert binary, not the php extension. This module states that other module ['Imagick'](https://www.drupal.org/project/imagick) requires the php extension (as an alternative).

My guess is we can discard the php extension, however right now we keep it in addition to the php modules installed as dependencies for imagemagick php module. These amount to 8MB, can test without if we care.

MaxCWhitehead committed:
Switch base image to ubi8-minimal.

- Use staged build pattern to discard build artifacts.
- Remove composer from this image.

nnewton commented on Dockerfile:
```diff
@@ -52,8 +79,4 @@ RUN { \
-
 USER 1001
-
-ENTRYPOINT [ "/usr/libexec/s2i/run" ]
```
Does this image not have an entry point atm then? It should be able to start on its own probably.

MaxCWhitehead replied:
Added entrypoint script to start apache and do some boilerplate configuration of apache/php/prefork mpm from environment variables.

MaxCWhitehead committed:
Add missing entrypoint/configuration of apache/php

Modify and create configuration in entrypoint script.
Switch user to apache from 1001 (not in ubi-minimal)

MaxCWhitehead commented:
Don't mind that force push, accidentally put something in and yanked it out.
